### ISSUE TITLE
[BUG] PGFPlotsX will not create output

### ISSUE THREAD

**Issue by gdbeck** [JuliaPlots/Plots.jl#2792]
## Details

PGFPlotsX does not seem to want to render figures as a backend to Plots.  I'm getting the following error:

```
julia> using Plots

julia> pgfplotsx()
Plots.PGFPlotsXBackend()

julia> plot(rand(10))
Error showing value of type Plots.Plot{Plots.PGFPlotsXBackend}:
ERROR: MethodError: Cannot `convert` an object of type Pair{String,RGBA{Float64}} to an object of type OrderedCollections.OrderedDict{Any,Any}
Closest candidates are:
  convert(::Type{OrderedCollections.OrderedDict{K,V}}, ::OrderedCollections.OrderedDict{K,V}) where {K, V} at /Users/gdb/.julia/packages/OrderedCollections/P6ntV/src/ordered_dict.jl:113
  convert(::Type{OrderedCollections.OrderedDict{K,V}}, ::AbstractDict) where {K, V} at /Users/gdb/.julia/packages/OrderedCollections/P6ntV/src/ordered_dict.jl:99
  convert(::Type{T}, ::T) where T<:AbstractDict at abstractdict.jl:486
  ...
Stacktrace:
 [1] PGFPlotsX.Options(::Pair{String,RGBA{Float64}}, ::Pair{String,Float64}) at /Users/gdb/.julia/packages/PGFPlotsX/gbWW3/src/options.jl:2
 [2] (::Plots.PGFPlotsXPlot)(::Plots.Plot{Plots.PGFPlotsXBackend}) at /Users/gdb/.julia/packages/Plots/ZVLhV/src/backends/pgfplotsx.jl:104
 [3] _update_plot_object(::Plots.Plot{Plots.PGFPlotsXBackend}) at /Users/gdb/.julia/packages/Plots/ZVLhV/src/backends/pgfplotsx.jl:1265
 [4] prepare_output(::Plots.Plot{Plots.PGFPlotsXBackend}) at /Users/gdb/.julia/packages/Plots/ZVLhV/src/plot.jl:196
 [5] display(::Plots.PlotsDisplay, ::Plots.Plot{Plots.PGFPlotsXBackend}) at /Users/gdb/.julia/packages/Plots/ZVLhV/src/output.jl:149
 [6] display(::Any) at ./multimedia.jl:323
 [7] #invokelatest#1 at ./essentials.jl:712 [inlined]
 [8] invokelatest at ./essentials.jl:711 [inlined]
 [9] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:161
 [10] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:146
 [11] (::REPL.var"#do_respond#38"{Bool,REPL.var"#48#57"{REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:729
 [12] #invokelatest#1 at ./essentials.jl:712 [inlined]
 [13] invokelatest at ./essentials.jl:711 [inlined]
 [14] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/LineEdit.jl:2354
 [15] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:1055
 [16] run_repl(::REPL.AbstractREPL, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:206
 [17] (::Base.var"#764#766"{Bool,Bool,Bool,Bool})(::Module) at ./client.jl:383
 [18] #invokelatest#1 at ./essentials.jl:712 [inlined]
 [19] invokelatest at ./essentials.jl:711 [inlined]
 [20] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:367
 [21] exec_options(::Base.JLOptions) at ./client.jl:305
 [22] _start() at ./client.jl:484
```

The error only occurs with pgfplotsx.  And I am able to use pgfplotsx directly.  For example, the following produces a pdf:

```
julia> using PGFPlotsX

julia> @pgf Axis(rand(10))

```

Any help is appreciated!

### Backends

This bug occurs on ( insert `x` below )

Backend      | yes | no  | untested
-------------|-----|-----|---------
gr (default) |     |  x   |
pyplot       |     |  x   |
plotly       |     |   x  |
plotlyjs     |     |     | x
pgfplotsx    |  x   |     |
inspectdr    |     |     | x

### Versions

Plots.jl version: 1.4.1
Backend  version (`]st -m`): [8314cec4] PGFPlotsX v1.2.7
Output of `versioninfo()`: 

Julia Version 1.4.2
Commit 44fa15b150* (2020-05-23 18:35 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.7.0)
  CPU: Intel(R) Core(TM) i7-6700K CPU @ 4.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-8.0.1 (ORCJIT, skylake)

**Comment by BeastyBlacksmith** [comment]
Hmm... I can't reproduce locally, but I also see that error on CI.
Can you reproduce this error, if you start julia with `julia --startup-file=no`?

**Comment by BeastyBlacksmith** [comment]
Okay, I can reproduce with PGFPlotsX v 1.2.7, so that is a downstream issue

**Comment by BeastyBlacksmith** [comment]
Can you confirm that it works, if you pin `PGFPlotsX` to version 1.2.6?

**Comment by gdbeck** [comment]
Pinning `PGFPlotsX` to version 1.2.6 fixes the problem.  Thank you so much for your help!

**Comment by BeastyBlacksmith** [comment]
Should be fixed with PGFPlotsX 1.2.8
